eww: unstable-2023-08-18 -> 0.5.0

coffeeispower opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

coffeeispower committed:
eww: unstable-2023-08-18 -> 0.5.0

eww: fix github src hash

eww: update cargo hash

Co-authored-by: a-kenji <aks.kenji@protonmail.com>

eww: use `rev = "v${version}"` instead of commit hash

Co-authored-by: kirillrdy <kirillrdy@gmail.com>

Nairou commented:
Release notes mentioned building both x11 and wayland support by default now. Is the `withWayland` flag still needed?

coffee-is-power commented:
Yeah i was thinking the same thing, we should deprecate the eww-wayland package

coffee-is-power commented:
Actually i will just do that without any waiting for any approval because this is pretty obvious that it is not needed anymore.

coffeeispower committed:
eww: deprecate eww-wayland package

eww: warn when using eww-wayland package

eww: move eww-wayland to `aliases.nix`

eww: use eww package in eww-wayland alias

eww: remove withWayland flag

adomixaszvers commented on pkgs/applications/window-managers/eww/default.nix:
```diff
@@ -5,32 +5,26 @@
+    hash = "sha256-06d8pm1cyg0zpa1kcnzzja2ia3c4glyr6ngkw1hs6l1pivj64kza";
   };
 
   cargoHash = "sha256-4yeu5AgleZMOLKNynGMd0XuyZxyyZ+RmzNtuJiNPN8g=";
```
You might update `cargoHash` if any of cargo dependencies were updated.

adomixaszvers replied:
use `lib.fakeHash` as it is written in nixpkgs manual about Rust packaging

a-kenji replied:
Yes, it needs to be updated:

> error: build of '/nix/store/qqr5gpd4fjr1pvm3m7zqgqjbck1bz6gl-eww-0.5.0-vendor.tar.gz.drv' on - failed: hash mismatch in fixed-output derivation '/nix/store/qqr5gpd4fjr1pvm3m7zqgqjbck1bz6gl-eww-0.5.0-vendor.tar.gz.drv':
         specified: sha256-4yeu5AgleZMOLKNynGMd0XuyZxyyZ+RmzNtuJiNPN8g=
            got:    sha256-IirFE714NZmppLjwbWk6fxcmRXCUFzB4oxOxBvmYu5U=

adomixaszvers commented on pkgs/applications/window-managers/eww/default.nix:
```diff
@@ -5,32 +5,26 @@
-    rev = "a9a35c1804d72ef92e04ee71555bd9e5a08fa17e";
-    hash = "sha256-GEysmNDm+olt1WXHzRwb4ZLifkXmeP5+APAN3b81/Og=";
+    rev = "387d344690903949121040f8a892f946e323c472";
+    hash = "sha256-06d8pm1cyg0zpa1kcnzzja2ia3c4glyr6ngkw1hs6l1pivj64kza";
```
This should be SRI hash but it looks like "sha256-" + some base32.

coffee-is-power replied:
idk how to generate that hash or cargoHash, i just tried using nix-prefetch-url on the tarball

SuperSandro2000 replied:
Set the hash to an empty string and run nix build and the hash will be printed.

adomixaszvers replied:
Or `nix-shell '<nixpkgs>' -p nix-prefetch-github --command 'nix-prefetch-github elkowar eww --rev v0.5.0'`

kirillrdy commented:
@coffee-is-power please restore PR template

also consult https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md regarding commit conventions

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -25813,6 +25813,8 @@ with pkgs;
 
   ## DEPRECATED, will be removed in a future release
 
+  eww-wayland = lib.warn "eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead." (callPackage ../applications/window-managers/eww {});
```
This must be in aliases.nix next to this file

coffee-is-power replied:
Did not notice that thanks

coffee-is-power replied:
fixed

coffee-is-power commented:
@kirillrdy I'll have to clone the entire repo to change the commit messages :skull: 
![imagem](https://github.com/NixOS/nixpkgs/assets/92828847/b27a03eb-e28c-4622-96d2-9a1685ef070b)

coffee-is-power commented:
So are we now ready to merge?

kirillrdy commented on pkgs/applications/window-managers/eww/default.nix:
```diff
@@ -5,32 +5,26 @@
     repo = "eww";
-    rev = "a9a35c1804d72ef92e04ee71555bd9e5a08fa17e";
-    hash = "sha256-GEysmNDm+olt1WXHzRwb4ZLifkXmeP5+APAN3b81/Og=";
+    rev = "387d344690903949121040f8a892f946e323c472";
```
```suggestion
    rev = "v${version}";
```

kirillrdy commented:
first commit message should be `eww: unstable-2023-08-18 -> 0.5.0` and so should be title of PR

kirillrdy reviewed (approved):
let's give others time to have a look

also small nitpick is to squash related commits ( eg hash updates, or change to wayland attribute alias etc )

SuperSandro2000 commented on pkgs/top-level/aliases.nix:
```diff
@@ -277,6 +277,7 @@ mapAliases ({
   eterm = throw "eterm was removed because it is still insecure: https://github.com/mej/Eterm/issues/7"; # Added 2023-09-10
   exa = throw "'exa' has been removed because it is unmaintained upstream. Consider using 'eza', a maintained fork"; # Added 2023-09-07
   exhibitor = throw "'exhibitor' has been removed because it is unmaintained upstream"; # Added 2023-06-20
+  eww-wayland = lib.warn "eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead." (callPackage ../applications/window-managers/eww {});
```
```suggestion
  eww-wayland = lib.warn "eww now can build for X11 and wayland simultaneously, so `eww-wayland` is deprecated, use the normal `eww` package instead." eww;
```

a-kenji commented:
Result of `nixpkgs-review pr 289595` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eww</li>
  </ul>
</details>

a-kenji reviewed (approved):
LGTM, pending the commit changes already requested.

coffee-is-power commented:
@a-kenji the commits are good now?

a-kenji commented:
Yes, LGTM.

Thanks for all the work!

coffee-is-power commented:
> Yes, LGTM.
> 
> Thanks for all the work!

btw it took ages to do, this repository is so big that my computer was dying to move and squash the commits with lazygit
